### PR TITLE
Raise error when inputs and mask don't have same shape in Softmax

### LINKED ISSUE #19722: softmax sliently generate a wrong output when the mask has an  incompatible shape
As shown in this [colab](https://colab.research.google.com/drive/1B5c-9QEwn8GWvi0kOp_mPvSt5BGV34tM?usp=sharing), the mask can arbitrarily change the shape of output.
Although the document requires that the mask has the same shape with input, no check exists  actually.
In my project, the shape of input is hard to compute and this potential incompatibility can affect the result and is hard to debug since softmax now silently generates an wrong output.
Currently the output is:

You can identify the shape changed by mask(line 2 and line 3 of output), which is different from the correct shape computed by compute_output_shape() (line4).

### PR DESCRIPTION
**Pull request by SuryanarayanaY**
Currently keras.layers.Softmax not having check to verify inputs and mask shape same or not. As per documnetation the mask should be A boolean mask of the same shape as inputs.

But Its generating output even though mask shape is not same as that of inputs.

Hence proposing a check to raise exception when inputs and mask don't have same shape.

Fixes #19722 .

### CODE REVIEW

File: keras/src/layers/activations/softmax.py [hunk 1]
```diff
@@ -50,6 +50,12 @@ def __init__(self, axis=-1, **kwargs):
 
     def call(self, inputs, mask=None):
         if mask is not None:
+            if mask.shape != inputs.shape:
+                raise ValueError(
+                    "`mask` and `inputs` must have same shape. "
```
[the remark this answers is not in the document]
Comment (SuryanarayanaY) [reply keras/src/layers/activations/softmax.py:55]: Actually documentation states that mask should be of same shape. May be we need to add here that same shape or broadcastable shape.
```
mask: A boolean mask of the same shape as `inputs`. The mask
          specifies 1 to keep and 0 to mask. Defaults to `None`.
```

Also the first dimension of shape i.e `batch_size` should not be considered in shape. Because if we pass `input_shape` as (5, 4, 1) and mask as (5,4) we get error as "`Incompatible shapes: [5,4,1] vs. [5,4]`". Do we need to mention this explicitly in docs?

Also the compute_output_shape method simply returns input_shape. Suppose if I pass 

```
x = np.random.rand(7, 1, 5)
mask = np.ones((5, 1))
l_1 = keras.layers.Softmax(axis=-1,trainable = True,dtype='float32',autocast = True)
print(l_1(x).shape) #(7, 1, 5)
print(l_1(x,mask).shape) #(7, 5,1) 
print(l_1.compute_output_shape(x.shape)) #(7, 1, 5)
```
Noticed output shape is different from `compute_output_shape` method as the `mask` is broadcastable here.
If I change mask shape to (1,5) instead of (5,1) then all shapes are same.

[the remark this answers is not in the document]
Comment (gbaned) [reply keras/src/layers/activations/softmax.py:55]: Hi @fchollet Any update on this PR? Please. Thank you!

[the remark this answers is not in the document]
Comment (gbaned) [reply keras/src/layers/activations/softmax.py:55]: Hi @fchollet Any update on this PR? Please. Thank you!

[the remark this answers is not in the document]
Comment (hertschuh) [reply keras/src/layers/activations/softmax.py:55]: Per this discussion, please revert the code change and instead update the documentation to state that the mask needs to be broadcastable to the input shape. Thanks!
